### PR TITLE
feat(snapshots): refactor snapshot openapi types

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**

### CODE REVIEW

File: control-plane/rest/openapi-specs/v0_api_spec.yaml [hunk 1]
```diff
@@ -3088,6 +3099,14 @@ components:
...
           type: string
           format: date-time
+        txn_id:
+          type: string
+        transactions:
+          type: object
+          additionalProperties:
+            type: array
+            items:
+              $ref: '#/components/schemas/ReplicaSnapshot'
```
Comment (Abhinandan-Purkait) [inline control-plane/rest/openapi-specs/v0_api_spec.yaml:3109]: Should these be `required` fields?

Comment (tiagolobocastro) [reply control-plane/rest/openapi-specs/v0_api_spec.yaml:3109]: `txn_id` and `transactions`? I think so.

**Review by tiagolobocastro — APPROVED** [review]

File: control-plane/rest/openapi-specs/v0_api_spec.yaml (hunk 1 above)
Comment (Abhinandan-Purkait) [reply control-plane/rest/openapi-specs/v0_api_spec.yaml:3109]: Yeah. Okay.

**Comment by Abhinandan-Purkait** [comment]
bors merge

**Pull request merged**